compute_tools/src/bin/compute_ctl.rs

```diff
@@ -370,6 +480,26 @@ fn main() -> Result<()> {
         exit_code = ecode.code()
     }
 
+    Ok(WaitPostgresResult { exit_code })
+}
+
+struct WaitPostgresResult {
+    exit_code: Option<i32>,
+}
+
+fn cleanup_and_exit(
```

Comment:
Splitting this function up into smaller logical chunks would also be great.

Reply:
Makes sense, yeah, will split it up a little bit more — hesitant to go much further though because IMO logically grouping actions by the "phase" of compute_ctl execution provides a lot of context when reading the code, and I worry that smaller chunks may leave `main()` as the only place with adequate context.

Reply:
Split it up further in 483559ae4741af296b68f1850d800b8e2eb37e6d